habitat-lab/habitat/tasks/rearrange/rearrange_grasp_manager.py

```diff
@@ -299,7 +304,10 @@ def update_object_to_grasp(self) -> None:
         is grasped then nothing will happen.
         """
 
-        if self._snapped_obj_id is None:
+        if (
+            self._snapped_obj_id is None
+            or not self._automatically_update_snapped_object
+        ):
```

Comment:
Then is automatically updating == this code handling the update?

Reply:
Correct.

Because this class is the mechanism for communicating the grasping state of each agent, this short circuit allows for external code to handle the transform updates.